src/main/java/org/clojars/gguthe/phpass/PHPass.java

```diff
@@ -80,10 +83,17 @@ private String cryptPrivate(String password, String setting) {
         if (((setting.length() < 2) ? setting : setting.substring(0, 2)).equalsIgnoreCase(output)) {
             output = "*1";
         }
-        String id = (setting.length() < 3) ? setting : setting.substring(0, 3);
+
+        String id = (setting.length() < 3) ? setting : setting.substring(0, 4);
+        if (id.equals("$2a$")) {
+            return BCrypt.hashpw(password, setting);

```

Comment:
The BCrypt.hashpw function doesn't do any base64 encoding.  This didn't interfere with the hash check, Is there some other reason to encode the hashes?